presto-main/src/main/java/com/facebook/presto/sql/planner/plan/PlanVisitor.java

```diff
@@ -16,9 +16,8 @@
 import com.facebook.presto.sql.planner.iterative.GroupReference;
 
 public abstract class PlanVisitor<R, C>
+        extends IRVisitor<R, C>
```

Comment:
I wound't use inheritance here. Let's have 2 separate `accept` methods.

Reply:
The inheritance is for visitors to visit both internal plan node as well as those to be moved to SPI (`FilterNode`, `TableScanNode`, etc). The plan nodes in SPI will only have one `accept` method. So we need to allow the visitor to reuse the `accept()` to visit them.